src/Symfony/Component/Config/Util/XmlUtils.php

```diff
@@ -104,6 +100,33 @@ public static function loadFile($file, $schemaOrCallable = null)
     }
 
     /**
+     * Loads an XML file.
+     *
+     * @param string               $file             An XML file path
+     * @param string|callable|null $schemaOrCallable An XSD schema file path, a callable, or null to disable validation
+     *
+     * @return \DOMDocument
+     *
+     * @throws \InvalidArgumentException When loading of XML file returns error
+     */
+    public static function loadFile($file, $schemaOrCallable = null)
+    {
+        $content = @file_get_contents($file);
+        if ('' === trim($content)) {
+            throw new \InvalidArgumentException(sprintf('File %s does not contain valid XML, it is empty.', $file));
+        }
+
+        try {
+            return static::parse($content, $schemaOrCallable);
+        } catch (\InvalidArgumentException $e) {
+            if ('The XML is not valid.' !== $e->getMessage()) {
```

Comment:
Logic sounds very fragile to me. Can we instead add a code to the exception and check the code here?